customer-managed/azure/terraform/network.tf

```diff
@@ -7,13 +12,19 @@ resource "azurerm_virtual_network" "redpanda" {
   tags = var.tags
 }
 
+data "azurerm_virtual_network" "redpanda" {
```

Comment:
super nit: Definitely personal preference, but I have been going the route of always creating the `data` object and using it when I need to reference values. This creates a more consistent pathway between the `create_vnet:false` and the `create_vnet:true` pathways.

```
data "azurerm_virtual_network" "redpanda" {
  name = var.vnet_name == "" ? azurerm_virtual_network.redpanda[0].name : var.vnet_name
  resource_group_name = azurerm_resource_group.network.name
}
```

This also eliminates one variable, no need for the `create_vnet` variable.